ProcessOptimizer/tests/test_space.py

```diff
@@ -677,31 +707,42 @@ def test_purely_categorical_space():
 # =============================================================================
 
 
-@pytest.mark.fast_test
-def test_lhs_arange():
-    # Test that the latin hypercube sampling functions run normally
-    dim_cat = Categorical(["a", "b", "c"])
-    dim_cat.lhs_arange(10)
-    dim_int = Integer(-10, 20)
-    lhs_int = dim_int.lhs_arange(10)
-    dim_real = Real(-10, 20)
-    lhs_real = dim_real.lhs_arange(10)
-    # Test that the Integer and Real classes return identical points after
-    # rounding, using Numpy's allclose function
-    assert np.allclose(lhs_int, lhs_real, atol=0.5)
-    # Test that the Integer class returns values that are ints for all intents 
-    # and purposes
-    assert all([np.mod(x,1) == 0 for x in lhs_int])
-    # Test that the Real class returns flots in all locations
-    assert all([isinstance(x,np.float64) for x in lhs_real])
-
-
 @pytest.mark.fast_test
 def test_lhs():
     SPACE = Space(
-        [Integer(-20, 20), Real(-10.5, 100), Categorical(list("abc"))]
+        [
+            Integer(1, 6),
+            Real(1, 7),
+            Real(10**-3, 10**3, prior="log-uniform"),
+            Categorical(list("abc")),
+        ]
     )
-    samples = SPACE.lhs(10)
-    assert len(samples) == 10
-    assert len(samples[0]) == 3
-
+    samples = SPACE.lhs(6)
+    assert len(samples) == 6
+    assert len(samples[0]) == 4
+    values = [[sample[i] for sample in samples] for i in range(4)]
+    # set disregards order, so we use it to test which values we got.
+    assert set(values[0]) == set(range(1, 6 + 1))
+    # We should get all six integers, in any order
+    assert set(values[1]) == set(0.5 + np.arange(1, 7))
+    # Six reals between 1 and 7, evenly divide, lie halfway between integers
```

Comment:
```suggestion
    # Six reals between 1 and 7, evenly divide, lie halfway between integers
    assert set(values[1]) == set(0.5 + np.arange(1, 7))
```
Moving comment up above the code it refers to for consistency.